src/node/proxy_agent.ts

```diff
@@ -0,0 +1,53 @@
+import { logger } from "@coder/logger"
+import * as http from "http"
+import * as proxyagent from "proxy-agent"
+
+/**
+ * This file does not have anything to do with the code-server proxy.
+ * It's for $HTTP_PROXY support!
+ * - https://github.com/cdr/code-server/issues/124
+ * - https://www.npmjs.com/package/proxy-agent
+ *
+ * This file exists in two locations:
+ * - src/node/proxy_agent.ts
+ * - lib/vscode/src/vs/base/node/proxy_agent.ts
+ * The second is a symlink to the first.
+ */
+
+/**
+ * monkeyPatch patches the node HTTP/HTTPS library to route all requests through our
+ * custom agent from the proxyAgent package.
+ */
+export function monkeyPatch(vscode: boolean): void {
+  if (!process.env.HTTP_PROXY) {
```

Comment:
Should we also take `HTTPS_PROXY`? VS Code also seems to use the lowercase versions, not sure if we should as well. https://github.com/microsoft/vscode/blob/1.51.1/src/vs/platform/request/node/proxy.ts#L11

Reply:
`process.env` is case insensitive! I think uppercase is canonical for environment variables so I think we should prefer that.

Yea I suppose we can just copy what VS Code's doing here.

Reply:
I might be confused actually, not sure if it's case sensitive or not.

Reply:
Ah yes, only on windows. I guess I'll just copy the VS Code function for this.

Reply:
Nvm, it's confusing. There's no difference between `HTTP_PROXY` and `HTTPS_PROXY` for us as `proxy-agent` just takes a single URL and automatically uses the right protocol for whatever is required. VS Code lets you separate the two protocols out but there's no real value in that as far as I can tell. So going to leave it as is with a comment and add the lowercase version.